tf-psa-crypto/tests/scripts/components-build-system.sh

```diff
@@ -0,0 +1,24 @@
+# components-build-system.sh
+#
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+
+# This file contains test components that are executed by all.sh
+
+################################################################
+#### Build System Testing
+################################################################
+
+component_test_cmake_tf_psa_crypto_out_of_source () {
+    msg "build: cmake tf-psa-crypto 'out-of-source' build"
+    TF_PSA_CRYPTO_ROOT_DIR="$PWD"
+    mkdir "$OUT_OF_SOURCE_DIR"
+    cd "$OUT_OF_SOURCE_DIR"
+    # Note: Explicitly generate files as these are turned off in releases
+    cmake -D CMAKE_BUILD_TYPE:String=Check -D GEN_FILES=ON "$TF_PSA_CRYPTO_ROOT_DIR"
```

Comment:
Echoing Harry's concern in https://github.com/Mbed-TLS/mbedtls/issues/9755#issuecomment-2460242562 : we currently prevent `component_test_cmake_out_of_source` from running on Ubuntu 16.04, but now `component_test_cmake_tf_psa_crypto_out_of_source` will run on Ubuntu 16.04. Why is this not a problem? I don't see it discussed in the review here.

The problem is a race condition. Just having a small number of successful CI runs is not enough to gain confidence.

Maybe the cmake file in tf-psa-crypto is better written and is not susceptible to the race condition. That's quite plausible, but do we actually know that?

Reply:
The component `component_test_cmake_tf_psa_crypto_out_of_source` is not added by this PR. This PR moves it from `tests/scripts/components-build-system.sh` to `tf-psa-crypto/tests/scripts/components-build-system.sh`. The component was added by the PR 9445 without protection against Ubuntu 16.04 (we just missed that point I think) which was merged October the 3rd. Thus it has been around for more that a month now. Enough to gain confidence?

Reply:
@gilles-peskine-arm I think your concern is valid in general, but quite unrelated to this PR, as Ronald explained.

> Feel free to override my change request if my concern is invalid.

Since you offered, I'm gonna do just that. I'm also going to start a discussion on slack about the potential issue and things we could do about it.

Reply:
Ok, I'd missed that it was a move and not new. I agree that a month of CI is enough to have confidence that this isn't a problem with the tf-psa-crypto cmake scripts.